### PR TITLE
Consistent Size Types

### PR DESCRIPTION
**Pull request by sslattery**
The purpose of this PR is to make the use of size types consistent in the code. I used the following conventions.

- If available, I used the template alias `size_type` associated with the Kokkos memory space in scope (see `AoSoA` or `Slice`)
- When a Kokkos memory space was not in scope, I used `std::size_t` for sizes

I think we should talk about my use of `std::size_t` in some cases (particularly `Tuple` and `SoA`) where we don't know the memory space in which we are working. Once choice is to use an integer template parameter for consistency. The other choice is to stick with my use of `std::size_t`.

### CODE REVIEW

**Comment by sslattery** [comment]
not ready for review yet

**Comment by sslattery** [comment]
OK this is ready for review

File: core/src/Cabana_SoA.hpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ struct InnerArrayType
   (including multidimensional arrays) as long as the type of T is trivial. A
   struct-of-arrays will be composed of these members of different types.
 */
```
[the remark this answers is not in the document]
Comment (sslattery) [reply core/src/Cabana_SoA.hpp]: You mean for the integer `M`?

File: core/src/Cabana_Slice.hpp [hunk 2]
```diff
@@ -561,170 +567,120 @@ class Slice
...
       \return The size of the array at the given struct index.
     */
-    template<typename S>
     KOKKOS_INLINE_FUNCTION
-    typename std::enable_if<std::is_integral<S>::value,int>::type
-    arraySize( const S& s ) const
+    size_type arraySize( const size_type s ) const
     {
-        return ( (std::size_t) s < _view.extent(0) - 1 )
+        return ( (size_type) s < _view.extent(0) - 1 )
```
[the remark this answers is not in the document]
Comment (sslattery) [reply core/src/Cabana_Slice.hpp]: Fixed - also found a few other c-style casts.

File: core/src/Cabana_SoA.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dalg24) [reply core/src/Cabana_SoA.hpp]: and the `get()` index arguments

[the remark this answers is not in the document]
Comment (sslattery) [reply core/src/Cabana_SoA.hpp]: Is there a change you want me to make? I guess I am still confused.

**Pull request merged**